include/meltpooldg/advection_diffusion/advection_diffusion_operation.hpp

```diff
@@ -21,7 +21,7 @@ namespace MeltPoolDG
   {
     using namespace dealii;
 
-    template <int dim, int comp = 0, int comp_hanging_node_constraints = comp + 1>
+    template <unsigned int dim>
```

Comment:
You are absolutely right
```suggestion
    template <int dim>
```
but let's stick with the old *inconsistency* of deal.II (to prevent in the future some automatic template detection issues).